state/protocol/badger/mutator.go

```diff
@@ -256,6 +278,14 @@ func (m *ParticipantState) Extend(ctx context.Context, candidate *flow.Block) er
 		return fmt.Errorf("could not check if block is an outdated extension: %w", err)
 	}
 
+	deferredBlockPersist := deferred.NewDeferredBlockPersist()
+
+	// check if the block header is a valid extension of parent block
+	err = m.headerExtend(ctx, candidate, nil, deferredBlockPersist)
+	if err != nil {
+		return fmt.Errorf("header not compliant with chain state: %w", err)
+	}
+
```

Comment:
for people reviewing my changes: 
* I have moved `headerExtend` below the calls [`checkBlockAlreadyProcessed`](https://github.com/onflow/flow-go/blob/b4951912cb40210a07018cd719987c2bf5b2de90/state/protocol/badger/mutator.go#L264) and [`checkOutdatedExtension`](https://github.com/onflow/flow-go/blob/b4951912cb40210a07018cd719987c2bf5b2de90/state/protocol/badger/mutator.go#L273)

thereby, we run the sanity checks first, before the more expensive business logic